### ISSUE TITLE
DTS API - support pagination and retrieval of entire text

### ISSUE THREAD

**Issue by balmas** [alpheios-project/alignment-editor-new#286]
Feedback from @PietroLiuzzo

1. support pagination of the collection response:

This is provided in the `view` property of the response:

```
"view" : {
    "first" : "/api/dts/collections?id=https://betamasaheft.eu/textualunits&page=1",
    "next" : "/api/dts/collections?id=https://betamasaheft.eu/textualunits&page=2",
    "last" : "/api/dts/collections?id=https://betamasaheft.eu/textualunits&page=204",
    "previous" : null,
    "@type" : "PartialCollectionView",
    "@id" : "/api/dts/collections?id=https://betamasaheft.eu/textualunits&page=1"
  },
```

2. support retrieval of entire text

`Resource` level collection members can provide a `dts:passage` property that allows you to retrieve the entire document rather than navigating the passages:

```
{
    ...
    "dts:passage" : "/api/dts/document?id=https://betamasaheft.eu/LIT4944History",
    "title" : "Nagara tǝsbǝʾtu la-walda ʾǝgziʾabǝḥer ʾǝgziʾǝna\n                    ʾIyasus Krǝstos",
    "@type" : "Resource",
    "@id" : "https://betamasaheft.eu/LIT4944History",
    "totalItems" : 0,
    "description" : "The abstract textual unit Nagara tǝsbǝʾtu la-walda ʾǝgziʾabǝḥer ʾǝgziʾǝna\n                    ʾIyasus Krǝstos in Beta maṣāḥǝft. ",
    "dts:download" : "https://betamasaheft.eu/tei/LIT4944History.xml",
    "dts:references" : "/api/dts/navigation?id=https://betamasaheft.eu/LIT4944History"
  },
```
It would be nice if we use this to offer the ability to import the entire text (of course of large documents this could present problems but the betamasaheft documents are all pretty small)

**Comment by irina060981** [comment]
Added pagination and ability to upload an entire document

![image](https://user-images.githubusercontent.com/12377640/113576349-c3bf7d00-9662-11eb-8081-6eabb7f69eb1.png)

![image](https://user-images.githubusercontent.com/12377640/113576425-e2be0f00-9662-11eb-987a-9253e36db3b7.png)

![image](https://user-images.githubusercontent.com/12377640/113576454-ebaee080-9662-11eb-925b-4d655f192a75.png)

**Comment by monzug** [comment]
Remote service is unavailable - https://betamasaheft.eu/api/dts/collections?id=https://betamasaheft.eu/narrativeunits for all 3 options so will try again tomorrow.

**Comment by PietroLiuzzo** [comment]
Sorry, I am unlikely to be able to look into why this does not work by tomorrow, but I will try and do it as soon as possible. Thanks!

**Comment by irina060981** [comment]
I was able to get data from
- https://betamasaheft.eu/api/dts/collections?id=https://betamasaheft.eu/textualunits
- https://betamasaheft.eu/api/dts/collections?id=https://betamasaheft.eu/transcriptions

And I failed to get data from
- https://betamasaheft.eu/api/dts/collections?id=https://betamasaheft.eu/narrativeunits

It was failed with the following from the application
```
Access to fetch at 'https://betamasaheft.eu/api/dts/collections?id=https://betamasaheft.eu/narrativeunits' from origin 'https://alpheios-misc-dev.s3.us-east-2.amazonaws.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

And when I tried to retrieve data with a direct link from browser - all 3 links were failed with 504 Error (Gateway Time-out)

After these fails all links stop to work - even this
https://betamasaheft.eu/api/dts/collections?id=https://betamasaheft.eu




![image](https://user-images.githubusercontent.com/12377640/113788853-48082200-9781-11eb-9a7e-81ccac90c4ac.png)

**Comment by PietroLiuzzo** [comment]
I have removed this collection since I cannot find out what is the issue. There are no texts there in any case.

**Comment by monzug** [comment]
I was able to browse the Beta maṣāḥǝft Textual Units. two issues here: as the units go from 1 to 204, wouldn't it be nice to have a 'go to'? e.g. to go directly to unit 104, I can just type the number 104.
also when I browse one unit (e.g. Ṣalot baʾǝnta ḥǝmāma wǝgāt) and I go back to the Beta maṣāḥǝft Textual Units, the unit numbers are not longer available. see screenshot
![Screen Shot 2021-04-07 at 4 22 10 PM](https://user-images.githubusercontent.com/41396793/113882851-f0d47280-97bd-11eb-805d-035690a724ec.png)

**Comment by monzug** [comment]
I think we should improve the following sentence: there are 2 ways to upload passages. in the following screenshot, there are actually 3 options for upload: entire document, 1 or 2. to make it clear I would say: you can either upload the entire document or just a single passage (or unit).

![Screen Shot 2021-04-07 at 4 33 53 PM](https://user-images.githubusercontent.com/41396793/113884621-6725a480-97bf-11eb-99fb-199db5ccb082.png)

**Comment by irina060981** [comment]
> also when I browse one unit (e.g. Ṣalot baʾǝnta ḥǝmāma wǝgāt) and I go back to the Beta maṣāḥǝft Textual Units, the unit numbers are not longer available. see screenshot

Fixed

**Comment by monzug** [comment]
as per You can either upload the entire document or just one or several passages:  maybe would be better as follow:
You can either upload the entire document or selected passages:
also, I noticed that when uploading the entire document is still visible.
![Screen Shot 2021-04-08 at 4 32 23 PM](https://user-images.githubusercontent.com/41396793/114045358-3fe6da00-9888-11eb-91bc-a98dcdb016e2.png)

**Comment by irina060981** [comment]
>I was able to browse the Beta maṣāḥǝft Textual Units. two issues here: as the units go from 1 to 204, wouldn't it be nice to have a 'go to'? e.g. to go directly to unit 104, I can just type the number 104.

added 
![image](https://user-images.githubusercontent.com/12377640/114122697-2d29df00-9934-11eb-9611-33c01b2f271b.png)

>You can either upload the entire document or selected passages:

updated

>also, I noticed that when uploading the entire document is still visible.

fixed

**Comment by monzug** [comment]
all fixed. Thanks. I would rather say "go to page" or "go to"

**Comment by irina060981** [comment]
updated title to `go to`
![image](https://user-images.githubusercontent.com/12377640/114333500-c7d02b00-9b8b-11eb-86d3-682d06fd28e6.png)

**Comment by monzug** [comment]
Thanks.
